api/capi/src/nntrainer.cpp

```diff
@@ -303,12 +303,48 @@ int ml_nnmodel_set_optimizer(ml_nnmodel_h model, ml_nnopt_h optimizer) {
   return status;
 }
 
+int ml_nnmodel_get_layer(ml_nnmodel_h model, const char *layer_name,
+                         ml_nnlayer_h *layer) {
+  int status = ML_ERROR_NONE;
+  ml_nnmodel *nnmodel;
+  ML_NNTRAINER_CHECK_MODEL_VALIDATION(nnmodel, model);
+
+  std::shared_ptr<nntrainer::NeuralNetwork> NN;
+  std::shared_ptr<nntrainer::Layer> NL;
+
+  std::unordered_map<std::string, ml_nnlayer *>::iterator layer_iter =
```

Comment:
I think this codes needs some description. It finds the layer:
1. layers_map
if not found : 2. find layers in NN
if not found : 3. make new layer and return ( just add new layer at the end of model?)
am i right?

Reply:
yes, using `ml_nnmodel_add_layer` is wrong. Entry should be only added to layers_map, and not to NeuralNetwork struct. Added #297 for bug fix and some description of this function.